### PR TITLE
Add launch file, exposure parameters, compressed image

### PR DESCRIPTION
**Pull request by jacobdeery**

### CODE REVIEW

File: src/ximea_driver.cpp [hunk 1]
```diff
@@ -6,20 +6,50 @@ namespace trailbot {
 
 constexpr int kTimeoutMs = 100;
 
-// Fixed exposure time
-// TODO(jacob): Make these parameters
-constexpr int kExposureMs = 50;
-constexpr int kExposureUs = kExposureMs * 1000;
-
 XIMEADriver::XIMEADriver() : Node("ximea_driver") {
-  pub_ = create_publisher<sensor_msgs::msg::Image>("camera", 10);
```
Comment (barzanisar) [inline src/ximea_driver.cpp:15]: Is there a reason why publisher is not created in the constructor?

File: src/ximea_driver.cpp [hunk 2]
```diff
@@ -6,20 +6,50 @@ namespace trailbot {
...
+  } else {
+    stat = xiSetParamInt(xi_handle_, XI_PRM_AEAG, XI_OFF);
+    const auto exposure_time_us = this->get_parameter("exposure_time_ms").as_int() * 1000;
+    stat = xiSetParamInt(xi_handle_, XI_PRM_EXPOSURE, exposure_time_us);
+    stat = xiSetParamFloat(xi_handle_, XI_PRM_GAIN, this->get_parameter("gain_db").as_double());
+  }
+
+  stat = xiSetParamFloat(xi_handle_, XI_PRM_GAMMAY, this->get_parameter("gamma_y").as_double());
+
+  stat = xiSetParamInt(xi_handle_, XI_PRM_HORIZONTAL_FLIP, XI_ON);
```
Comment (barzanisar) [inline src/ximea_driver.cpp:59]: What do these params mean? Can you please add comments?

File: src/ximea_driver_node.cpp [hunk 3]
```diff
@@ -4,7 +4,8 @@ using namespace trailbot;
 
 int main(int argc, char** argv) {
   rclcpp::init(argc, argv);
-  rclcpp::spin(std::make_shared<XIMEADriver>());
```
Comment (barzanisar) [inline src/ximea_driver_node.cpp:7]: Interesting that cpp ros node does not need spinning

File: src/ximea_driver.cpp [hunk 4]
```diff
@@ -6,20 +6,50 @@ namespace trailbot {
...
+    stat = xiSetParamInt(xi_handle_, XI_PRM_AEAG, XI_ON);
+    stat = xiSetParamFloat(xi_handle_, XI_PRM_EXP_PRIORITY, 0.5);
+
+    const auto max_exposure_time_us = this->get_parameter("max_exposure_time_ms").as_int() * 1000;
+    stat = xiSetParamInt(xi_handle_, XI_PRM_AE_MAX_LIMIT, max_exposure_time_us);
+
+    const auto target_brightness = this->get_parameter("target_brightness").as_int();
+    stat = xiSetParamInt(xi_handle_, XI_PRM_AEAG_LEVEL, target_brightness);
+  } else {
+    stat = xiSetParamInt(xi_handle_, XI_PRM_AEAG, XI_OFF);
```
Comment (barzanisar) [inline src/ximea_driver.cpp:48]: what does this param do?

File: src/ximea_driver.cpp [hunk 5]
```diff
@@ -6,20 +6,50 @@ namespace trailbot {
...
+
   stat = xiSetParamInt(xi_handle_, XI_PRM_BUFFER_POLICY, XI_BP_SAFE);
   stat = xiSetParamInt(xi_handle_, XI_PRM_IMAGE_DATA_FORMAT, XI_MONO8);
 
-  stat = xiSetParamInt(xi_handle_, XI_PRM_EXPOSURE, kExposureUs);
+  const bool auto_exposure = this->get_parameter("auto_exposure").as_bool();
+
+  if (auto_exposure) {
+    stat = xiSetParamInt(xi_handle_, XI_PRM_AEAG, XI_ON);
+    stat = xiSetParamFloat(xi_handle_, XI_PRM_EXP_PRIORITY, 0.5);
```
Comment (barzanisar) [inline src/ximea_driver.cpp:39]: what is this param?

**Review by barzanisar — CHANGES_REQUESTED** [review]
Looks good. Just need some comments to explain what some parameters mean.

File: src/ximea_driver.cpp (hunk 1 above)
Comment (jacobdeery) [reply src/ximea_driver.cpp:15]: The publisher is created from the `ImageTransport`, which requires the node to be properly initialized, which only happens after the constructor is complete.

File: src/ximea_driver.cpp (hunk 5 above)
Comment (jacobdeery) [reply src/ximea_driver.cpp:39]: Will add comments.

File: src/ximea_driver.cpp (hunk 2 above)
Comment (jacobdeery) [reply src/ximea_driver.cpp:59]: Wilco

File: src/ximea_driver.cpp (hunk 4 above)
Comment (jacobdeery) [reply src/ximea_driver.cpp:48]: Commented

File: src/ximea_driver_node.cpp (hunk 3 above)
Comment (jacobdeery) [reply src/ximea_driver_node.cpp:7]: There aren't any callbacks, so no need to spin

**Pull request merged**